### PR TITLE
 task/[FLCRM-14331] | Adds sort param for toTileSQL 

### PR DESCRIPTION
**Pull request by canriquezds**
This PR will update the `toTileSQL` method to take a new parameter, `sorting`.
With this change,  fulcrum app will have the flexibility to pass to tiles service the sorting field and direction currently applied to the table within the mixed view on the `data-viewer`.

Why:

JIRA Ticket: [`FLCRM-14331 | WEB: Set Z-index of records on map according to sort order in settings`](https://fulcrumapp.atlassian.net/browse/FLCRM-14331)

Next steps:
This PR is the base for the next ticket  [FLCRM-15221 | [fulcrum] Set Z-index of records on map according to sort order in settings](https://fulcrumapp.atlassian.net/browse/FLCRM-15221) that uses the updated `toTileSQL` method to apply the sorting to the rendered tiles layer on the ESRI map.

## Testing: Running environment against branch with:
Access to the preview:
[Testing preview | rails8842.thefulcrum.team](https://rails8842.thefulcrum.team/admin)
- Login as owner in this orga: [1_Seeded Esri Organization. Safe to use!](https://rails8842.thefulcrum.team/admin/customers/657d204f-364e-4c69-b325-a6eeba4fab47)
- Create an app with advanced geometries. ( Or access this [app: OSMv3](https://rails8842.thefulcrum.team/dash/c84503ab-21b2-4cff-9de1-f6bf5f31c43a)) Enable title using the for example the `name` field
- Create overlapping records
- Apply filter to the table on the split view
- Verify the geometries order changes also in the map

### Video demo

https://github.com/fulcrumapp/fulcrum-query-sql/assets/94479185/9aab8d12-c9db-4bdf-a933-4725741518b6

### CODE REVIEW

**Review by JuanAcosta112 — APPROVED** [review]
Works great! 💯 
<p align="center">
Screenshot 2024-05-09 at 12 05 55 PM
</p>

**Review by JuanAcosta112 — APPROVED** [review]

**Review by corporealshift — DISMISSED** [review]

**Review by JuanAcosta112 — DISMISSED** [review]

**Review by corporealshift — DISMISSED** [review]

**Review by JuanAcosta112 — APPROVED** [review]

**Comment by hdaof** [comment]
Should unit tests be added to account for sorting? Perhaps in `src/ast/__tests__/converter.test.js` or create a new test file?

**Comment by canriquezds** [comment]
image

**Review by JuanAcosta112 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -32,3 +32,45 @@ yarn build
...
+
+```
+yarn remove fulcrum-query-sql
+```
+- After that
+
+```
+yarn add fulcrum-query-sql@github:fulcrumapp/fulcrum-query-sql#BRANCH-NAME
+```
+2. Clean Yarn Cache: Clear the yarn cache to remove any stored data from previous installs which might conflic
```
Comment (hdaof) [inline README.md:53]:
```suggestion
2. Clean Yarn Cache: Clear the yarn cache to remove any stored data from previous installs which might conflict
```

**Review by hdaof — DISMISSED** [review]

**Review by hdaof — APPROVED** [review]

**Review by JuanAcosta112 — APPROVED** [review]

**Pull request merged**